pkg/ebpftracer/policy_filters.go

```diff
@@ -216,6 +176,96 @@ func DeduplicateDnsEvents(l *logging.Logger, size uint32, ttl time.Duration) Eve
 	}
 }
 
+func DnsEventsFilter(log *logging.Logger, size uint32, ttl time.Duration) PreEventFilterGenerator {
+	type cacheValue struct{}
+
+	return func() PreEventFilter {
+		cache, err := freelru.New[uint64, cacheValue](size, func(key uint64) uint32 {
+			return uint32(key) //nolint:gosec
+		})
+		// err is only ever returned on configuration issues. There is nothing we can really do here, besides
+		// panicing and surfacing the error to the user.
+		if err != nil {
+			panic(err)
+		}
+
+		cache.SetLifetime(ttl)
+		var discard uint8
+
+		return func(ctx *types.EventContext, decoder *decoder.Decoder) (types.Args, error) {
+			if ctx.EventID != events.NetPacketDNSBase {
+				return nil, FilterPass
+			}
+
+			// Read firsts two bytes and discard. It's mapped to argsnum and index.
+			// For network events in most cases there is only 1 argument (payload).
+			_ = decoder.DecodeUint8(&discard)
```

Comment:
This is not good imho. We now leak details about how to parse the data from the kernel outside the `decoder` package.

Reply:
Moved most of the logic to decoder